data/projects/optimize_caching.md

```diff
@@ -1,28 +1,50 @@
-This project involves improving coala's performance by implementing caching
-strategies. One example is to use git-like approach at file-collection level
-to not explore directories that haven't been modified. Caching per section can
-also be implemented so sections without issues don't rerun on all files where
-issues in other sections occurred.
+Currently, coala supports caching of project files. This was a big performance
+improvement over running code analysis for all files, even if most files
+are unchanged since the last run.
 
-Own performance optimization ideas are highly encouraged!
+In this project, you're to improve on the current strategies. Some enhancements
+such as section-level caching, globbing improvements to not expand on
+unmodified directories, and others have been preliminarily explored. Please
+refer to the related issues to learn more about these.
 
+The student is also highly encouraged to come up with newer solutions that may
+increase performance even further. The applicant is allowed to flexibly change
+the milestones depending on how major the newly proposed solutions are, after
+explicit approval from the mentors and admin.
 
 #### Milestones
 
 ##### GSOC 2017 COMMUNITY BONDING
 
-* The applicant has determined a number of circumstances where coala needlessly
-checks unmodified files.
-* The applicant has created testcases to validate performance improvements
-* A cEP describing the changes is merged.
+* The applicant has identified at least 3 other performance hotspots that can be  
+  improved. These can be smaller improvements, focussing on one particular  
+  aspect of coala's core.
+* The applicant has drafted and merged a cEP describing all the proposed  
+  changes thoroughly.
+* Preliminary tests have been performed to validate the proposed changes'  
```

Comment:
No, this is more - you hack up a rough implementation (without tests, docs, and just the bare features) and test if it works decently. This is mostly a litmus test for testing whether your solution is sane - mostly to convince yourself that the proposed changes *might* (might, not will) work.